sqlite3_test.go

```diff
@@ -1722,6 +1723,44 @@ func TestAuthorizer(t *testing.T) {
 	}
 }
 
+func TestFileControl(t *testing.T) {
+	t.Run("PERSIST_WAL", func(t *testing.T) {
+		tempFilename := TempFilename(t)
+		defer os.Remove(tempFilename)
+
+		sql.Register("sqlite3_FCNTL_PERSIST_WAL", &SQLiteDriver{
+			ConnectHook: func(conn *SQLiteConn) error {
+				arg := int(1)
+				if rc := conn.FileControl("", SQLITE_FCNTL_PERSIST_WAL, unsafe.Pointer(&arg)); rc != 0 {
+					t.Fatalf("Unexpected return code from FileControl(): %d", rc)
```

Comment:
You probably should not call `t.Fatalf` here, since this maybe happening on some other goroutine, depending how database/sql works. Instead, just return an error. Below, call `db.Ping` after `sql.Open` and move the assertion there.